src/pages/index.js

```diff
@@ -1,81 +1,199 @@
 function createCard(data) {
-  const cardElement = new Card(data, constants.cardSelector, handleImageClick);
+  const cardElement = new Card(
+    data,
+    constants.cardSelector,
+    handleImageClick,
+    handleLikeClick,
+    handleDeleteClick
+  );
   return cardElement.getView();
 }
-// Section for rendering cards
-const cardSection = new Section(
-  {
-    items: constants.initialCards,
-    renderer: renderCard,
-  },
-  ".cards__list"
-);
-cardSection.renderItems();
-
-function renderCard(item) {
-  const cardElement = createCard(item);
-  cardSection.addItem(cardElement);
-}
-// UserInfo
+// UserInfo and Cards
 const userInformation = new UserInfo({
   name: ".profile__title",
   description: ".profile__description",
+  avatar: ".profile__image",
 });
 
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  // UserInfo
+  .then(([userData, cards]) => {
+    userInformation.setUserInfo({
+      name: userData.name,
+      description: userData.about,
+    });
+    userInformation.setAvatar(userData.avatar);
+    // Cards
+    cardSection = new Section(
+      {
+        items: cards,
+        renderer: (cardData) => {
+          const cardEl = createCard(cardData);
+          cardSection.addItem(cardEl);
+        },
+      },
+      ".cards__list"
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+// PopupWithImage and image preview
+const popupImage = new PopupWithImage("#modal-preview");
+popupImage.setEventListeners();
+
+function handleImageClick(data) {
+  popupImage.open(data);
+}
+// Likes
+function handleLikeClick(cardId) {
+  if (cardId._isLiked) {
+    api
+      .removeLike(cardId._id)
+      .then(() => {
+        cardId._renderLikes();
+        cardId._isLiked = false;
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .addLike(cardId._id)
+      .then(() => {
+        cardId._renderLikes();
+        cardId._isLiked = true;
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
 // PopupWithForm
-const addCardModal = new PopupWithForm("#add-modal", handleAddCardSubmit);
-addCardModal.setEventListeners();
+const avatarModal = new PopupWithForm("#avatar-modal", handleAvatarSubmit);
+avatarModal.setEventListeners();
 
 const profileEditModal = new PopupWithForm("#edit-modal", handleEditSubmit);
 profileEditModal.setEventListeners();
 
+const addCardModal = new PopupWithForm("#add-modal", handleAddCardSubmit);
+addCardModal.setEventListeners();
 // Handlers
-function handleImageClick(data) {
-  popupImage.open(data);
+function handleAvatarSubmit({ link }) {
+  avatarModal.renderLoad(true);
+  api
+    .updateAvatar(link)
+    .then((res) => {
+      userInformation.setAvatar(res.avatar);
+      avatarModal.close();
+      constants.avatarForm.reset();
+      avatarFormValidator.toggleBtnState();
+    })
+    .catch((err) => {
+      console.error("Oops...Failed to update avatar:", err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);`